core/cmd/on_chain_config_cmd.go

```diff
@@ -0,0 +1,399 @@
+				return nil, RiverError(Err_INVALID_ARGUMENT, "invalid uint64 value", "value", strNum)
+			}
+			nums[i] = num
+		}
+		return crypto.ABIEncodeUint64Array(nums), nil
+	default:
+		return nil, RiverError(Err_INVALID_ARGUMENT, "invalid value type", "type", valueType)
+	}
+}
+
+type setArgs struct {
+	key      string
+	blockNum uint64
+	value    []byte
+}
+
+func parseSetArgs(args []string, force bool) (setArgs, error) {
+	if len(args) < 3 || len(args) > 4 {
+		return setArgs{}, RiverError(
+			Err_INVALID_ARGUMENT,
+			"need key, blockNum, value, and optionally type",
+			"len",
+			len(args),
+		)
+	}
+	key := args[0]
+	knownType, ok := crypto.AllKnownOnChainSettingKeys()[key]
+	if !ok {
+		if !force {
+			return setArgs{}, RiverError(Err_INVALID_ARGUMENT, "key is not known", "key", key)
+		}
+	}
+	blockNumStr := args[1]
+	value := args[2]
+	var valueType string
+	if len(args) > 3 {
+		valueType = args[3]
+	}
+
+	if valueType == "" {
+		if knownType == "" {
+			return setArgs{}, RiverError(Err_INVALID_ARGUMENT, "need explicit type for key", "key", key)
+		}
+		valueType = knownType
+	} else if !force && knownType != valueType {
+		return setArgs{}, RiverError(Err_INVALID_ARGUMENT, "type mismatch for key", "key", key, "known_type", knownType, "provided_type", valueType)
+	}
+
+	blockNum, err := strconv.ParseUint(blockNumStr, 10, 64)
+	if err != nil {
+		return setArgs{}, err
+	}
+
+	valueBytes, err := encodeValue(valueType, value)
+	if err != nil {
+		return setArgs{}, err
+	}
+
+	return setArgs{
+		key:      key,
+		blockNum: blockNum,
+		value:    valueBytes,
+	}, nil
+}
+
+func submitConfig(ctx context.Context, cfg *config.Config, args []setArgs) error {
+	wallet, err := crypto.NewWalletFromEnv(ctx, "PRIVATE_KEY")
+	if err != nil {
+		return err
+	}
+
+	blockchain, err := crypto.NewBlockchain(
+		ctx,
+		&cfg.RiverChain,
+		wallet,
+		infra.NewMetricsFactory(nil, "river", "cmdline"),
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+	blockchain.StartChainMonitor(ctx)
+
+	caller, err := river.NewRiverConfigV1Transactor(cfg.RegistryContract.Address, blockchain.Client)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	var txs []crypto.TransactionPoolPendingTransaction
+	for _, arg := range args {
+		fmt.Printf("Setting %s to %s on block %d\n", arg.key, hex.EncodeToString(arg.value), arg.blockNum)
+		tx, err := blockchain.TxPool.Submit(
+			ctx,
+			"SetConfiguration",
+			func(opts *bind.TransactOpts) (*types.Transaction, error) {
+				return caller.SetConfiguration(
+					opts,
+					crypto.HashSettingName(arg.key),
+					arg.blockNum,
+					arg.value,
+				)
+			},
+		)
+		if err != nil {
+			return err
+		}
+		txs = append(txs, tx)
+	}
+
+	for _, tx := range txs {
+		receipt, err := tx.Wait(ctx)
+		if err != nil {
+			return err
+		}
+		if receipt.Status != types.ReceiptStatusSuccessful {
+			return RiverError(Err_INTERNAL, "transaction failed", "tx", receipt.TxHash)
```

Comment:
Maybe capture all transaction results and return a summary instead of returning immediately so there is an overview which transaction did fail and which pass.